sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1878,17 +1878,25 @@ class Dataset[T] private[sql](
   def dropDuplicates(colNames: Seq[String]): Dataset[T] = withTypedPlan {
     val resolver = sparkSession.sessionState.analyzer.resolver
     val allColumns = queryExecution.analyzed.output
-    val groupCols = colNames.map { colName =>
-      allColumns.find(col => resolver(col.name, colName)).getOrElse(
+    val groupCols = colNames.flatMap { colName =>
+      // It is possibly there are more than one columns with the same name,
+      // so we call filter instead of find.
+      val cols = allColumns.filter(col => resolver(col.name, colName))
+      if (cols.isEmpty) {
         throw new AnalysisException(
```

Comment:
My thought is:

When an user mistakenly gives wrong column to `Dataset.drop`, it can be easily found out.

But for `Dataset.dropDuplicates`, it might be harder to figure out duplicate rows are still there. So to throw an explicit exception looks more proper to me.